src/compiler/checker.ts

```diff
@@ -10273,13 +10276,19 @@ namespace ts {
                 }
             }
 
+            if (stringIndexTypes.length || numberIndexTypes.length) {
+                members.forEach((symbol, key) => {
+                    (isNumericLiteralName(key) ? numberIndexTypes : stringIndexTypes).push(getTypeOfSymbol(symbol));
+                });
+            }
+
             const spread = createAnonymousType(
                 symbol,
                 members,
                 emptyArray,
                 emptyArray,
-                getNonReadonlyIndexSignature(stringIndexInfo),
-                getNonReadonlyIndexSignature(numberIndexInfo));
+                leftStringIndexInfo || rightStringIndexInfo ? createIndexInfo(getUnionType(stringIndexTypes.concat(numberIndexTypes), UnionReduction.Subtype), /*readonly*/ false) : undefined,
+                leftNumberIndexInfo || rightNumberIndexInfo ? createIndexInfo(getUnionType(numberIndexTypes, UnionReduction.Subtype), /*readonly*/ false) : undefined);

```

Comment:
this always loses the declaration of the original signature. is this information necessary to retain?